taca/server_status/server_status.py

```diff
@@ -158,17 +158,47 @@ def get_uppmax_cpu_hours():
     for proj in output.strip().split('\n'):
         project_dict = {"time": current_time.isoformat()}
 
-        # split line into a list
-        project = proj.split()
+        try: # split line into a list
+            project_id, cpu_hours, cpu_limit = proj.split()
         # sometimes it returns empty strings or something strange
-        try:
-            # creating objects
-            project_dict["project"] = project[0]
-            project_dict["cpu hours"] = project[1]
-            project_dict["cpu limit"] = project[2]
-            result[project[0]] = project_dict
         except Exception, e:
             logging.error(e.message)
-            continue
+        else:
+            project_dict["project"] = project_id
+            project_dict["cpu hours"] = cpu_hours
+            project_dict["cpu limit"] = cpu_limit
+
+            quotas = _get_uppmax_cpu_quotas(project_id)
+            if quotas:
+                project_dict["cpu_quota_decrease"] = quotas
+
+            result[project_id] = project_dict
+
+    return result
 
+def _get_uppmax_cpu_quotas(project_id):
+    today = datetime.date.today()
+    result = {}
+    try:
+        logging.info("CPU Quotas Decrease, project {}:".format(project_id))
+        cpu_quotas = subprocess.Popen("grep -A 10 {} /sw/uppmax/etc/projects".format(project_id).split(), stdout=subprocess.PIPE)
```

Comment:
Need a different command here - `-A 10` will get the next ten lines after the project name. But there can be any number of these lines, and they will grow over time. So can't rely on this, instead need to parse properly.

Something like this maybe? (Pseudocode, not tested)
```python
with open ('/sw/uppmax/etc/projects, "r") as f:
  in_proj = False
  for l in f:
    if project_id in l:
      in_proj = True
    if in_proj and l.startswith('Shortgrants:'):
      logging.info(line)
      try:
        shortgrants, cpu_hours, date_string = line.split()
# [.. etc ..]
```